### PR TITLE
store/tikv: add RC isolation level.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: kv/kv.go [hunk 1]
```diff
@@ -147,6 +149,8 @@ type Request struct {
 	// ResponseIterator.Next is called. If concurrency is greater than 1, the request will be
 	// sent to multiple storage units concurrently.
 	Concurrency int
+	// IsolationRC sets the isolation level as RC(Read Committed).
+	IsolationRC bool
```
Comment (AndreMouche) [inline kv/kv.go]: Why not make the isolation level as an enum?

File: store/tikv/mock-tikv/mvcc.go [hunk 2]
```diff
@@ -95,11 +95,9 @@ func (e *mvccEntry) lockErr() error {
 	}
 }
 
-func (e *mvccEntry) Get(ts uint64) ([]byte, error) {
-	if e.lock != nil {
-		if e.lock.startTS <= ts {
-			return nil, e.lockErr()
-		}
+func (e *mvccEntry) Get(ts uint64, isoLevel kvrpcpb.IsolationLevel) ([]byte, error) {
+	if e.lock != nil && e.lock.startTS <= ts && isoLevel == kvrpcpb.IsolationLevel_SI {
```
Comment (AndreMouche) [inline store/tikv/mock-tikv/mvcc.go]: Could we check `isoLevel == kvrpcpb.IsolationLevel_SI ` first like `tikv` do?

File: store/tikv/snapshot.go [hunk 3]
```diff
@@ -114,6 +117,9 @@ func (s *tikvSnapshot) batchGetKeysByRegions(bo *Backoffer, keys [][]byte, colle
 }
 
 func (s *tikvSnapshot) batchGetSingleRegion(bo *Backoffer, batch batchKeys, collectF func(k, v []byte)) error {
+	sender := NewRegionRequestSender(s.store.regionCache, s.store.client)
+	sender.SetIsolationLevel(s.isolationLevel)
```
Comment (AndreMouche) [inline store/tikv/snapshot.go]: I prefer to put `isolationLevel ` in `NewRegionRequestSender`, so we can easily check whether we have missed a `sender.SetIsolationLevel(s.isolationLevel)`?

**Comment by disksing** [comment]
PTAL @AndreMouche

**Comment by nolouch** [comment]
LGTM

**Comment by AndreMouche** [comment]
@zhangjinpeng1987 @hhkbp2 PTAL

File: store/tikv/mock-tikv/mvcc.go [hunk 4]
```diff
@@ -95,9 +95,9 @@ func (e *mvccEntry) lockErr() error {
 	}
 }
 
-func (e *mvccEntry) Get(ts uint64) ([]byte, error) {
-	if e.lock != nil {
-		if e.lock.startTS <= ts {
+func (e *mvccEntry) Get(ts uint64, isoLevel kvrpcpb.IsolationLevel) ([]byte, error) {
+	if isoLevel == kvrpcpb.IsolationLevel_SI {
+		if e.lock != nil && e.lock.startTS <= ts {
 			return nil, e.lockErr()
```
Comment (zhangjinpeng1987) [inline store/tikv/mock-tikv/mvcc.go]: Why not `if isoLevel == kvrpcpb.IsolationLevel_SI && if e.lock != nil && e.lock.startTS <= ts`?

Comment (disksing) [inline store/tikv/mock-tikv/mvcc.go]: To keep consistent with tikv.

**Pull request merged**